pkg/apis/numaflow/v1alpha1/deprecated.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2022 The Numaproj Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"os"
+	"strconv"
+)
+
+// TODO: (k8s 1.29) Remove this once we deprecate the support for k8s < 1.29
+func isSidecarSupported() bool {
```

Comment:
I see the EnvK8sServerVersion is set when we start reconciller. Will it ever be possible that we call isSidecarSupported before reconciller cmd start?

Reply:
Good question. I think it's only used in the controller, should be okay.